lib/react_client.dart

```diff
@@ -844,6 +895,25 @@ dynamic _findDomNode(component) {
       component is Component ? component.jsThis : component);
 }
 
+class ReactDartContext {
+  dynamic _jsThis;
+  ReactDartContext(this.Provider, this.Consumer, this._jsThis);
+  ReactJsComponentFactoryProxy Provider;
+  ReactJsComponentFactoryProxy Consumer;
+  dynamic get jsThis => _jsThis;
+}
+
+ReactDartContext createContext(
+    [dynamic defaultValue, Function calculateChangedBits]) {
```

Comment:
Same here, can we make this typed?

Reply:
```dart
ReactDartContext createContext<T>([
    T defaultValue,
    int Function(T, T) calculateChangedBits,
]);
```

@greglittlefield-wf @kealjones-wk do you agree it makes sense to parameterize `createContext` to ensure that `defaultValue` and the type of the args passed to the `calculateChangedBits` function are the same?

Reply:
The javascript in my blood doesn't agree, but i understand why you guys want to make it this way... :'(